### PR TITLE
[PIO-183] Add Jupyter Docker image

### PR DESCRIPTION
**Pull request by marevol**

### CODE REVIEW

**Comment by marevol** [comment]
I pushed docker images.
You can try [JUPYTER.md](https://github.com/marevol/predictionio/blob/jupyter/docker/JUPYTER.md) without building them.
Please let me know if you have any questions/problems.

**Comment by marevol** [comment]
I pushed docker images.
You can try [JUPYTER.md](https://github.com/marevol/predictionio/blob/jupyter/docker/JUPYTER.md) without building them.
Please let me know if you have any questions/problems.

**Comment by marevol** [comment]
I pushed docker images.
You can try [JUPYTER.md](https://github.com/marevol/predictionio/blob/jupyter/docker/JUPYTER.md) without building them.
Please let me know if you have any questions/problems.

**Comment by marevol** [comment]
I pushed docker images.
You can try [JUPYTER.md](https://github.com/marevol/predictionio/blob/jupyter/docker/JUPYTER.md) without building them.
Please let me know if you have any questions/problems.

File: docker/jupyter/start.sh [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+if [ $(grep _MYSQL_ $PIO_ENV_FILE | wc -l) = 0 ] ; then
+  sed -i "s/^MYSQL/#MYSQL/" $PIO_ENV_FILE
+fi
+
+# start event server
+sh /usr/bin/pio_run &
+
+export PYSPARK_PYTHON=$CONDA_DIR/bin/python
+export PYSPARK_DRIVER_PYTHON=$CONDA_DIR/bin/jupyter
+export PYSPARK_DRIVER_PYTHON_OPTS=notebook
```
Comment (shimamoto) [inline docker/jupyter/start.sh]: There are duplicate processes in `start-jupyter.sh`. Are both necessary?

File: docker/JUPYTER.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+unset PYSPARK_PYTHON
+unset PYSPARK_DRIVER_PYTHON
+unset PYSPARK_DRIVER_PYTHON_OPTS
+```
+
+To create a model, run:
+
+```
+pio train --main-py-file train.py
+```
```
Comment (shimamoto) [inline docker/JUPYTER.md]: What do you think it is preferable to clear the above environment variables when running `pio train` in Python?

File: docker/JUPYTER.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+limitations under the License.
+-->
+
+Jupyter With PredictionIO
+=========================
+
+## Overview
+
+Using Jupyter based docker, you can use Jupyter Notebook with PredictionIO environment.
+It help your exploratory data analysis (EDA).
```
[the remark this answers is not in the document]
Comment (marevol) [reply docker/JUPYTER.md]: Fixed.

File: docker/JUPYTER.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+The above command prints a token to the console as below.
+
+```
+pio_1       |         http://(3aaf67361022 or 127.0.0.1):8888/?token=e87a634b4ab7e2c8bcd86aea9def3eb48183c043eac86f3e
+```
+
+Open `http://127.0.0.1:8888/`, type the token, and then open a new terminal in Jupyter from `New` pulldown button.
+
+## Getting Started With Scala Based Template
```
[the remark this answers is not in the document]
Comment (marevol) [reply docker/JUPYTER.md]: Yes. For scale template, you can run pio command and build it in Jupyter's terminal.

File: docker/jupyter/start.sh (hunk 1 above)
Comment (marevol) [reply docker/jupyter/start.sh]: Removed.

File: docker/JUPYTER.md (hunk 2 above)
Comment (marevol) [reply docker/JUPYTER.md]: This Jupyter has PYSPARK_* environment variables. 
If these variables are not removed, `pio train` uses them and starts Jupyter in Jupyter...
So, we need to clear them before executing `pio train` in Jupyter terminal.

File: docker/JUPYTER.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (shimamoto) [reply docker/JUPYTER.md]: Wow, that's great! I'll give it a try.

File: docker/JUPYTER.md (hunk 2 above)
Comment (shimamoto) [reply docker/JUPYTER.md]: Exactly, so I have come up with a following plan: if the `main-py-file` option is specified, PYSPARK_* environment variables are removed automatically before running `spark-submit` internally. Each has its pros and cons. It doesn't have to be implemented :)

Comment (marevol) [reply docker/JUPYTER.md]: We can set settings other than Jupyter ones to PYSPARK_*.
I think that it's better not to clear them in pio command...

**Comment by marevol** [comment]
I'll merge this PR this week. Please let me know if you have any comments.

**Pull request merged**